plonky2/src/hash/arch/aarch64/poseidon_goldilocks_neon.rs

```diff
@@ -52,9 +40,12 @@ const FINAL_ROUND_CONSTANTS: [u64; WIDTH * HALF_N_FULL_ROUNDS] = make_final_roun
 const fn check_mds_matrix() -> bool {
     // Can't == two arrays in a const_assert! (:
     let mut i = 0;
-    let wanted_matrix_exps = [0, 0, 1, 0, 3, 5, 1, 8, 12, 3, 16, 10];
+    let wanted_matrix_circ = [17, 15, 41, 16, 2, 28, 13, 13, 39, 18, 34, 20];
+    let wanted_matrix_diag = [8, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
```

Comment:
I feel you might have explained this before, but why is the circulant matrix above the (almost) reverse of the matrix I originally gave: `[17, 20, 34, 18, 39, 13, 13, 28, 2, 16, 41, 15]`?

Reply:
Oh,  I guess the transpose has been taken somewhere.

Reply:
Yeah, the difference is whether you're storing the first row or the first column of a circulant matrix. The generic Poseidon code stores the first row. But to do convolution in Fourier space, you actually need to take the FFT of the first column. So it's the matrix you gave me, I just define it by storing a different thing.